services/api/src/resources/organization/resolvers.ts

```diff
@@ -242,6 +242,30 @@ export const getOrganizationById: ResolverFn = async (
     return orgResult;
 };
 
+export const getOrganizationByName: ResolverFn = async (
+  organization,
+  args,
+  { sqlClientPool, hasPermission }
+) => {
+  let orgName = args.organization;
+  if (organization) {
+    orgName = organization;
+  }
+
+  await hasPermission('organization', 'view', {
+    organization: orgName,
```

Comment:
this won't work as the attribute has to be the organization id, not the name

you'll need to run the sql query prior to the haspermission check to retrieve the `orgResult.id` to pass to haspermissions

Reply:
basically shift all of this above haspermissions
```
  const rows = await query(sqlClientPool, Sql.selectOrganizationByName(orgName));
  const orgResult = rows[0];

  if (!orgResult) {
    return null;
  }
```

Reply:
Ah yeah of course, done!